vault-core/src/main/java/org/apache/jackrabbit/vault/packaging/registry/impl/FSPackageRegistry.java

```diff
@@ -231,7 +231,7 @@ private File getPackageMetaDataFile(@Nonnull PackageId id) {
     /**
      * {@inheritDoc}
      */
-    public VaultPackage open(File pkg) throws RepositoryException {
+    public VaultPackage open(File pkg) throws IOException {
```

Comment:
I would not return NULL here, as the consumer (FSRegisteredPackage) doesn't expect this.